### PR TITLE
[Enhancement] YAML validation of a TestStep file

### PR DESCRIPTION
**Pull request by shubham-cmyk**
Fixes https://github.com/kyverno/kuttl/issues/24

- If a nonexisting assert, apply, or error file is present in the test step it will fail
- If TestStep contains any random field ( foo ) it will fail
- It would check the yaml indentation on TestStep, assert files, error files, and apply files. ( Strict decoder )

### CODE REVIEW

**Comment by shubham-cmyk** [comment]
@eddycharly Please Check

**Comment by eddycharly** [comment]
The right approach would be to use strict unmarshalling.
As in https://github.com/kyverno/kyverno/blob/173bb907b7380292460879c5de80d946a305cda8/cmd/cli/kubectl-kyverno/values/load.go#L18

**Comment by chipzoller** [comment]
Can you please use code blocks rather than screenshots of terminal output?

**Comment by shubham-cmyk** [comment]
@chipzoller @eddycharly 
Can you check ?

**Comment by shubham-cmyk** [comment]
- If a nonexisting assert, apply, or error file is present in the test step it will fail
- If TestStep contains any random field ( foo )  it will fail
- It would check the yaml indentation on TestStep, assert files, error files, and apply files. ( Strict decoder )

File: pkg/test/step.go [hunk 1]
```diff
@@ -561,6 +564,14 @@ func (s *Step) LoadYAML(file string) error {
...
 		if apply.object.GetObjectKind().GroupVersionKind().Kind == "TestStep" {
 			if testStep, ok := apply.object.(*harness.TestStep); ok {
+				// Validate TestStep
+				if err := validateTestStep(testStep, s.Dir); err != nil {
+					return fmt.Errorf("failed to validate TestStep object from %s: %v", file, err)
+				}
+				// Check for YAML errors in the TestStep file
+				if err := checkYAMLError(file); err != nil {
+					return fmt.Errorf("failed to check YAML in TestStep file %s: %v", file, err)
+				}
```
Comment (eddycharly) [inline pkg/test/step.go:574]: But you already decoded the yaml file here right ?

Comment (shubham-cmyk) [reply pkg/test/step.go:574]: File could be of kind=TestStep

Below are the indentation check of the files that are present in the apply,assert and error

Comment (shubham-cmyk) [reply pkg/test/step.go:574]: I don't think here the func validateTestStep() Check of the kind=TestStep as it is only concerned about the apply,assert,error

Comment (eddycharly) [reply pkg/test/step.go:574]: If we can write `apply.object.(*harness.TestStep)`, that probably means we decode the yaml already.

Comment (shubham-cmyk) [reply pkg/test/step.go]: Now used : `FromUnstructuredWithValidation(unstruct, converted, true)`

**Review by eddycharly — APPROVED** [review]
LGTM, thanks !

**Pull request merged**